### PR TITLE
Add non-canonical output sorting.

### PR DESCRIPTION
**Pull request by litghost**

### CODE REVIEW

**Comment by litghost** [comment]
Example output: https://gist.github.com/litghost/c28544821468903d046ddd3615465c8c

File: fasm/output.py [hunk 1]
```diff
@@ -0,0 +1,312 @@
+import enum
```
Comment (mithro) [inline fasm/output.py]: File needs a module level docstring.

File: fasm/output.py [hunk 2]
```diff
@@ -0,0 +1,312 @@
...
+def is_only_annotation(line):
+    """ Returns True if line is only an annotations. """
+    return not line.set_feature and line.annotations and not line.comment
+
+def is_blank_line(line):
+    """ Returns True if line is blank. """
+    return not line.set_feature and not line.annotations and not line.comment
+
+def merge_features(features):
+    """ Combines features with varying addresses but same feature into one feature.
```
Comment (mithro) [inline fasm/output.py]: These examples could be better as doctests?

File: fasm/output.py [hunk 3]
```diff
@@ -0,0 +1,312 @@
...
+        NoGroup = 1
+        InCommentGroup = 2
+        InAnnotationGroup = 3
+
+    def __init__(self):
+        self.state = MergeModel.State.NoGroup
+        self.groups = []
+        self.current_group = None
+
+    def start_comment_group(self, line):
```
Comment (mithro) [inline fasm/output.py]: (optional) These functions could use doc strings.

**Comment by mithro** [comment]
Should the example output you posted just be a unittest?

File: fasm/output.py (hunk 2 above)
Comment (litghost) [reply fasm/output.py]: Doctests are the wrong answer given how much is printed.  pytests would be good, but not in this PR.

**Comment by litghost** [comment]
> Can we log a bug to add auto-formatting tools like prjxray?

Done #8

**Comment by litghost** [comment]
> Should the example output you posted just be a unittest?

That output actually requires the prjxray database for the sort and zero functions, so it would be hard.

**Comment by mithro** [comment]
```
py27 runtests: commands[2] | flake8 .
./fasm/output.py:93:80: E501 line too long (80 > 79 characters)
./fasm/output.py:95:80: E501 line too long (84 > 79 characters)
./fasm/output.py:330:80: E501 line too long (80 > 79 characters)
ERROR: InvocationError for command '/home/travis/build/SymbiFlow/fasm/.tox/py27/bin/flake8 .' (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   py27: commands failed
The command "tox" exited with 1.
```

File: fasm/output.py (hunk 3 above)
Comment (litghost) [reply fasm/output.py]: Added comments for the public ones.

**Comment by litghost** [comment]
> ```
> py27 runtests: commands[2] | flake8 .
> ./fasm/output.py:93:80: E501 line too long (80 > 79 characters)
> ./fasm/output.py:95:80: E501 line too long (84 > 79 characters)
> ./fasm/output.py:330:80: E501 line too long (80 > 79 characters)
> ERROR: InvocationError for command '/home/travis/build/SymbiFlow/fasm/.tox/py27/bin/flake8 .' (exited with code 1)
> ___________________________________ summary ____________________________________
> ERROR:   py27: commands failed
> The command "tox" exited with 1.
> ```

Fixed, maybe.

File: fasm/output.py (hunk 2 above)
Comment (litghost) [reply fasm/output.py]: #9

**Pull request merged**